src/libs/ease/reference.c

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file
+ *
+ * @brief Reference methods.
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include "kdbease.h"
+
+#include "kdbhelper.h"
+
+#include <string.h>
+
+
+/**
+ * Check whether a reference is redundant (i.e. it can be expressed in less characters) or not.
+ *
+ * @param reference the reference to check
+ * @retval 1 if the reference is redundant
+ * @retval 0 otherwise
+ */
+int elektraIsReferenceRedundant (const char * reference)
```

Comment:
Do we really want to export this API call?

Can you give an example how it can be used by an user?

And some tests would be nice.

Reply:
It isn't really useful for applications, but it can be used by plugins and other parts of elektra.
The `reference` plugin for example produces a warning, if a reference is redundant, because this is probably a mistake. The same could be done by any other future plugin that uses `references` (default value calculation, better conditionals, etc.).

> And some tests would be nice.

I thought about adding test, but we would basically just test `keySetName/keyAddName`. The 3 `if`s inside `elektraResolveReference` directly correspond to the documentation above and `elektraIsReferenceRedundant` is also very very simple.

Also, `testmod_reference` from the `reference` plugin will fail, if this method is broken. Because the methods originally where contained in the plugin, all currently possible cases should be covered already.

coveralls also reports [100% coverage](https://coveralls.io/builds/22003479/source?filename=src/libs/ease/reference.c)

Reply:
> It isn't really useful for applications, but it can be used by plugins and other parts of elektra...

I did not doubt that this is useful to have in a lib. Only the docu did not explain what you are writing here.

> coveralls also reports 100% coverage

Ok, then it is called indirectly.